packages/stepper/src/InternalStep.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { PropsWithChildren } from 'react';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { StepStates, InternalStepProps } from './types';
+import StepIcon from './StepIcon';
+import { palette } from '@leafygreen-ui/palette';
+import { spacing } from '@leafygreen-ui/tokens';
+import { useStepperContext } from './StepperContext';
+import StepLabel from './StepLabel';
+import { stepIconClassName } from './constants';
+
+const Step = ({
+  children,
+  index,
+  state,
+  ariaLabel = `step${index || ''}`,
+  shouldDisplayLine = true,
+  iconSize = 20,
+  className,
+  ...rest
+}: PropsWithChildren<InternalStepProps & React.HTMLProps<HTMLDivElement>>) => {
+  const { isDarkMode } = useStepperContext();
+  const isCurrent = state === StepStates.Current;
+  const isCompleted =
+    state === StepStates.Completed || state === StepStates.CompletedMultiple;
+
+  const baseStyles = css`
+    display: flex;
+    margin: auto;
+    flex-direction: column;
+    align-items: center;
+    padding-bottom: ${spacing[1]}px;
+    position: relative; // for the :after line
+
+    &:focus-visible {
+      outline: none;
+      .${stepIconClassName} {
+        // TODO: should use box-shadow utility for this.
+        box-shadow: 0px 0px 0px 2px
+            ${isDarkMode ? palette.black : palette.white},
+          0px 0px 0px 4px ${palette.blue.light1};
+      }
+    }
+
+    ${shouldDisplayLine &&
+    css`
+      &:after {
+        content: '';
+        height: 1px;
+        width: 100%;
+        position: absolute;
+        top: ${iconSize / 2}px;
+        left: 50%;
+        z-index: 0;
+        background-color: ${isDarkMode
+          ? palette.gray.light1
+          : palette.gray.base};
+      }
+    `}
+  `;
```

Comment:
⚠️  Should add this conditionally with `cx` instead of interpolating a boolean here

Reply:
We don't really need to be conservative with cssString variables—they don't add much overhead (and I'm unclear as to how this would drastically increase the number of vars in this case).
We've established a pattern where if we have a block of CSS that is toggled by some prop, it should be added using `cx` to make it clear what's toggling the styles.

Reply:
OH I accidentally thought you were referring to the `isDarkMode` ternary. Using `cx` based on the props makes sense to me. I'll update this.

Reply:
From our conversation yesterday, this looks fine 👍. I'd prefer this to be outside the component for performance & maintenance but that doesn't need to block this PR